src/components/vr-scatter-plot/vr-scatter-plot.component.ts

```diff
@@ -1,32 +1,95 @@
+import { LineChartMeta } from '../../datasets/metas/line-chart.meta';
+import { BehaviorSubject, Subject } from 'rxjs';
+import { DAY } from '../../utils/timeUnits';
+import { map, takeUntil } from 'rxjs/operators';
 
 @Component({
   selector: 'app-vr-scatter-plot',
   templateUrl: './vr-scatter-plot.component.html'
 })
-export class VRScatterPlotComponent implements OnInit, OnChanges, OnDestroy{
+export class VRScatterPlotComponent<T extends Meta>implements OnInit, OnChanges, OnDestroy{
+  @Input() endDate = new Date();
+  @Input() startDate = new Date(this.endDate.getTime() - 30 * DAY);
+  @Input() meta: Meta;
+  @Input() meta2: LineChartMeta;
   private vrScatterPlot: Scatterplot;
-  private shape: string;
-  private color: string;
+  componentMetas: Meta[];
+  queryOptions$ = new BehaviorSubject<TimeSeriesQueryOptions>({
+    range: [this.startDate, this.endDate],
+  });
+  datum$ = new BehaviorSubject<LineChartDatum>({
+    label: '',
+    points: [],
+  });
+  private destroy$ = new Subject();
 
   constructor(
+    private preferenceService: PreferenceService
   ) {
-    this.vrScatterPlot = new Scatterplot('a-sphere');
+    this.preferenceService = preferenceService;
   }
+
   ngOnInit() {
-   this.vrScatterPlot.init(document.querySelector('a-scene'), [10, 20, 30, 40, 50, 60]);
+    this.vrScatterPlot = new Scatterplot('a-sphere');
+    const dataService = new DataService(this.preferenceService);
+    dataService.dataset$
+    .pipe(takeUntil(this.destroy$))
+    .subscribe(dataset => {
+      // componentMetas is initialized to different dataset metas - will help funnel dataset
+      this.componentMetas = dataset.metas;
+      // dataset.metas[0].type = 'tabbed' and dataset.metas[1] = 'line'
+      this.meta = dataset.metas[0];
+      if (this.isTabbed()) {
+        this.meta2 = (this.meta as any).metas[0];
+        console.log(this.meta2);
+      } else {
+        this.meta2 = this.meta[0];
+      }
+      // calling this.init() from inside bc when code above is in constructor,
+      // and code in this.init() is in ngOnInit() there is sync problem
+      this.init();
+    });
   }
 
   ngOnDestroy() {
   }
 
   ngOnChanges(changes: SimpleChanges): void {
-    // if ('measureName' in changes) {
-    //   this.data = this.dataService.getMeasureOverDays(this.measureName);
-    //   this.activeDatum = null;
-    }
   }
+
+  init(){
+    this.queryOptions$
+    .pipe(takeUntil(this.destroy$))
+    .pipe(map(queryOption => {
+      // this.meta2.query(queryOption)[0]) has label, points, style and is of type BehaviorSubject<LineChartData>
+      return this.meta2.queryData(queryOption)[0];
+    }))
+    .subscribe(this.datum$);
+    this.vrScatterPlot.init(document.querySelector('a-scene'), this.datum$.value.points);
+  }
+
+  isTabbed(): boolean {
+    return 'tabbed' === this.meta.type;
+  }
+
+  get datum() {
+    return this.datum$.value;
+  }
+
+  sleep(milliseconds: number) {
```

Comment:
Why is this needed? Can you use `setTimeout` instead?

Reply:
I used this during debugging because I there was a timing/sync problem -- I'm not super sure, but from what I read on component life cycles in Angular the `constructor` and `onInit` functions could be running at the same time... however for my design I need everything in the constructor to be declared/initialized before `onInit` is called.

My solution was to put the meta handling of `DataService` in `onInit` and then at the end of the function, I called the second chunk of code, where I access `queryOptions` of the meta selected in `onInit`. This worked, but please let me know if you have any suggestions!

Reply:
`constructor` should be called prior to `ngOnInit `. 

Two thoughts here that might help:
1. You could wrap the contents of `init` in a `setTimeout` call with a small delay.
2. You could try using another lifecycle hook (e.g. `ngAfterViewInit`) that runs later, instead of `ngOnInit`.  See: https://angular.io/guide/lifecycle-hooks